### ISSUE TITLE
【upgrade】ks3.3.1->ks3.4.0, system components tower is locked

### ISSUE THREAD

**Issue by luqiaoshi** [kubesphere/kubesphere#5855]
**Describe the Bug**
**Pre:** ks 3.3.1+k8s v1.21.14
**Steps:**
1.download kk v3.0.9
2../kk upgrade --with-kubernetes v1.23.17 --with-kubesphere v3.4.0
**Actual:**
system components tower is locked

```
kubesphere-system              ks-apiserver-7b9cb45797-6qpkh                              1/1     Running     0               3h30m
kubesphere-system              ks-apiserver-7b9cb45797-7hvhs                              1/1     Running     0               3h29m
kubesphere-system              ks-apiserver-7b9cb45797-nc785                              1/1     Running     0               3h28m
kubesphere-system              ks-console-6c64b9cb79-284pd                                1/1     Running     0               3h30m
kubesphere-system              ks-console-6c64b9cb79-rt27w                                1/1     Running     0               3h29m
kubesphere-system              ks-console-6c64b9cb79-zfdz7                                1/1     Running     0               3h29m
kubesphere-system              ks-controller-manager-5b5495c5f9-7c27m                     1/1     Running     3 (3h28m ago)   3h29m
kubesphere-system              ks-controller-manager-5b5495c5f9-8f98l                     1/1     Running     4 (167m ago)    3h30m
kubesphere-system              ks-controller-manager-5b5495c5f9-nnjmv                     1/1     Running     1 (167m ago)    3h28m
kubesphere-system              ks-installer-757d576db7-96ttr                              1/1     Running     0               3m
kubesphere-system              minio-b7b9cf94-zmjhc                                       1/1     Running     0               6h20m
kubesphere-system              openldap-0                                                 1/1     Running     1 (6h17m ago)   6h20m
kubesphere-system              openldap-1                                                 1/1     Running     1 (5h49m ago)   6h17m
kubesphere-system              openpitrix-import-job-4q4rq                                0/1     Completed   0               102s
kubesphere-system              redis-74f975c764-s7mck                                     1/1     Running     1 (18h ago)     23h
kubesphere-system              tower-8645f8b7bf-8lssd                                     1/1     Running     0               6h56m
kubesphere-system              tower-bc76945f7-s7zw7                                      0/1     Running     0               18s
```
view pod logs:
```
root@node1:~# kubectl -n kubesphere-system      logs -f        tower-bc76945f7-s7zw7
I0721 07:48:31.193092       1 options.go:46] CA set to "/ca.crt".
I0721 07:48:31.193127       1 options.go:47] CA key file set to "/ca.key".
I0721 07:48:31.193131       1 options.go:48] Host set to 0.0.0.0
I0721 07:48:31.193134       1 options.go:49] Agent port set to 8080.
I0721 07:48:31.193142       1 options.go:50] Kubeconfig set to "".
I0721 07:48:31.193151       1 options.go:51] Leader election set to true
W0721 07:48:31.193374       1 client_config.go:552] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
I0721 07:48:31.194366       1 leaderelection.go:242] attempting to acquire leader lease  kubesphere-system/tower...
I0721 07:48:31.200988       1 leaderelection.go:341] lock is held by tower-8645f8b7bf-8lssd_b17a246d-a5e9-4adf-8655-0c8bf0c42089 and has not yet expired
I0721 07:48:31.201001       1 leaderelection.go:247] failed to acquire lease kubesphere-system/tower
I0721 07:48:39.832462       1 leaderelection.go:341] lock is held by tower-8645f8b7bf-8lssd_b17a246d-a5e9-4adf-8655-0c8bf0c42089 and has not yet expired
I0721 07:48:39.832478       1 leaderelection.go:247] failed to acquire lease kubesphere-system/tower
I0721 07:48:50.478981       1 leaderelection.go:341] lock is held by tower-8645f8b7bf-8lssd_b17a246d-a5e9-4adf-8655-0c8bf0c42089 and has not yet expired
I0721 07:48:50.478995       1 leaderelection.go:247] failed to acquire lease kubesphere-system/tower
I0721 07:48:59.469107       1 leaderelection.go:341] lock is held by tower-8645f8b7bf-8lssd_b17a246d-a5e9-4adf-8655-0c8bf0c42089 and has not yet expired
I0721 07:48:59.469122       1 leaderelection.go:247] failed to acquire lease kubesphere-system/tower
I0721 07:49:07.100451       1 leaderelection.go:341] lock is held by tower-8645f8b7bf-8lssd_b17a246d-a5e9-4adf-8655-0c8bf0c42089 and has not yet expired
I0721 07:49:07.100467       1 leaderelection.go:247] failed to acquire lease kubesphere-system/tower
I0721 07:49:14.652930       1 leaderelection.go:341] lock is held by tower-8645f8b7bf-8lssd_b17a246d-a5e9-4adf-8655-0c8bf0c42089 and has not yet expired
I0721 07:49:14.652944       1 leaderelection.go:247] failed to acquire lease kubesphere-system/tower
I0721 07:49:23.777980       1 leaderelection.go:341] lock is held by tower-8645f8b7bf-8lssd_b17a246d-a5e9-4adf-8655-0c8bf0c42089 and has not yet expired
I0721 07:49:23.777994       1 leaderelection.go:247] failed to acquire lease kubesphere-system/tower
I0721 07:49:29.175143       1 leaderelection.go:341] lock is held by tower-8645f8b7bf-8lssd_b17a246d-a5e9-4adf-8655-0c8bf0c42089 and has not yet expired
```

**Versions Used**
KubeSphere: `ks v3.4.0`
Kubernetes: (If KubeSphere installer used, you can skip this)

**Environment**
How many nodes and their hardware configuration:

For example: CentOS 7.5 / 3 masters:  8cpu/8g; 3 nodes: 8cpu/16g
(and other info are welcomed to help us debugging)

**How To Reproduce**

**Expected behavior**

**Comment by iawia002** [comment]
Due to improper leader election configuration, the new Pod cannot be fully started, and the old Pod cannot be deleted, so it is stuck in a dead loop. As a temporary solution, you can change the replicas of the deployment to 0 and then back to 1.

**Comment by luqiaoshi** [comment]
issue fixed in `kse v3.4.1.230807 `
```
kubesphere-system              ks-apiserver-858d6f796f-bbr45                      1/1     Running             0               8m57s
kubesphere-system              ks-apiserver-858d6f796f-ldsmm                      0/1     ContainerCreating   0               4m12s
kubesphere-system              ks-apiserver-858d6f796f-p62lh                      1/1     Running             0               7m5s
kubesphere-system              ks-console-6f754b85c8-dnkfv                        1/1     Running             0               8m57s
kubesphere-system              ks-console-6f754b85c8-mpz59                        0/1     ImagePullBackOff    0               7m4s
kubesphere-system              ks-console-c59ddcdc-qs548                          1/1     Running             0               156m
kubesphere-system              ks-controller-manager-5cf86b66d5-g46vn             0/1     ContainerCreating   0               5m35s
kubesphere-system              ks-controller-manager-5cf86b66d5-gtp5g             1/1     Running             1               8m24s
kubesphere-system              ks-controller-manager-5cf86b66d5-spm4k             1/1     Running             1 (3m31s ago)   8m57s
kubesphere-system              ks-installer-5d7c6bbfc5-w5kvs                      1/1     Running             0               13m
kubesphere-system              minio-859cb4d777-zqccc                             1/1     Running             0               159m
kubesphere-system              openpitrix-import-job-v6rfz                        0/1     Completed           0               8m29s
kubesphere-system              redis-68d7fd7b96-c8rc2                             1/1     Running             0               160m
kubesphere-system              tower-548c7654b7-sgkck                             1/1     Running             0               7m52s
```

**Comment by luqiaoshi** [comment]
issue fixed in `v3.4.0`
```
kubesphere-system              ks-apiserver-869d75cb7d-cwdcl                      1/1     Running   0               8m35s
kubesphere-system              ks-apiserver-869d75cb7d-cwhgv                      1/1     Running   0               9m3s
kubesphere-system              ks-apiserver-869d75cb7d-ztfjf                      1/1     Running   0               9m31s
kubesphere-system              ks-console-57b987d8b-66gg7                         1/1     Running   0               9m31s
kubesphere-system              ks-console-57b987d8b-9bjkt                         1/1     Running   0               9m4s
kubesphere-system              ks-console-57b987d8b-ks7kt                         1/1     Running   0               9m19s
kubesphere-system              ks-controller-manager-7d4f8fb564-9t9pq             1/1     Running   4 (8m1s ago)    9m31s
kubesphere-system              ks-controller-manager-7d4f8fb564-x7kw9             1/1     Running   0               7m7s
kubesphere-system              ks-controller-manager-7d4f8fb564-zsbxh             1/1     Running   1 (3m56s ago)   6m49s
kubesphere-system              ks-installer-6b7fbd66cd-b5mgr                      1/1     Running   0               12m
kubesphere-system              redis-d744b7468-lnmrn                              1/1     Running   0               97m
kubesphere-system              tower-5c56d94b84-nvssq                             1/1     Running   0               8m36s
```
